src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,120 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+import static duke.common.Messages.MESSAGE_FAREWELL;
+import static duke.common.Messages.MESSAGE_GREETING;
+import static duke.common.Messages.MESSAGE_INQUIRY;
+import static duke.common.Messages.MESSAGE_ADD_TASK_ACKNOWLEDGEMENT;
+import static duke.common.Messages.MESSAGE_DELETE_TASK_ACKNOWLEDGEMENT;
+import static duke.common.Messages.MESSAGE_MARK_TASK_ACKNOWLEDGEMENT;
+import static duke.common.Messages.MESSAGE_UNMARK_TASK_ACKNOWLEDGEMENT;
+import static duke.common.Messages.MESSAGE_TASK_LISTING_HEADER;
+
+public class Ui {
+    private static final String DIVIDER = "____________________________________________________________";
+
+    private Scanner in;
+
+    public Ui() {
+        this.in = new Scanner(System.in);
+    }
+
+    //@@author wcwy-reused
+    //Reused from from https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/ui/TextUi.java
+    // with minor modifications
+    public void displayMessages(String... messages) {
+        for (String message : messages) {
+            System.out.println(message);
+        }
+    }
+    //@@author
+
+    public String readCommand() {
+        String userInput = in.nextLine();
+        return userInput;
+    }
+
+    public void displayGreetingMessage() {
+        displayMessages(
+                DIVIDER,
+                MESSAGE_GREETING,
+                MESSAGE_INQUIRY,
+                DIVIDER
+        );
+    }
+
+    public void displayListingMessage(String listContent) {
+        displayMessages(
+                DIVIDER,
+                MESSAGE_TASK_LISTING_HEADER,
+                listContent,
+                DIVIDER
+        );
+    }
+
+    public void displayTaskAdditionMessage(String taskDetails, int count) {
+        displayMessages(
+                DIVIDER,
+                MESSAGE_ADD_TASK_ACKNOWLEDGEMENT,
+                "\t" + taskDetails,
+                "Now you have " + count + " tasks in the list",
+                DIVIDER
+        );
+    }
+
+    public void displayTaskDeletionMessage(String taskDetails, int count) {
+        displayMessages(
+                DIVIDER,
+                MESSAGE_DELETE_TASK_ACKNOWLEDGEMENT,
+                "\t" + taskDetails,
+                "Now you have " + count + " tasks in the list",
+                DIVIDER
+        );
+    }
+
+    public void displayMarkOrUnmarkMessage(String taskName, boolean isMark) {
+        displayMessages(
+                DIVIDER,
+                getMessageAcknowledgementForMarkAndUnmark(isMark),
+                getCheckboxForMarkAndUnmark(isMark) + taskName,
+                DIVIDER
+        );
+    }
```

Comment:
It's error-prone to carry this sort of boolean (like `isMark`) as a parameter to implement effectively different methods under the same name. If it needs to be done, it should be, if possible, restricted at the top-level, e.g. `displayMarkOrUnmarkMessage` takes `isMark` as a parameter, but `getMessageAcknowledgementForMarkAndUnmark` and `getCheckboxForMarkAndUnmark` are refactored into two methods each and do not.

In general, you might want to think how to refactor the whole thing to not needed the boolean parameter. What is the high-level operation you are trying to achieve? Write methods for that.

Reply:
Thanks for sharing this! I wasn't aware that it was error-prone to refractor the code this way, now knowing this, I split the displayMessage for mark and unmark back into this:

```
    public void displayTaskMarkedMessage(String taskName) {
        displayMessages(
                DIVIDER,
                MESSAGE_MARK_TASK_ACKNOWLEDGEMENT,
                FORMATTED_MARK_LOGO + taskName,
                DIVIDER
        );
    } 
    // Similar for unmark
```
The code now does not carry the boolean values around anymore and the number of methods used is even reduced from 3 to 2, with far shorter code.